ml-agents/mlagents/trainers/learn.py

```diff
@@ -396,6 +397,12 @@ def create_environment_factory(
             .replace(".x86_64", "")
             .replace(".x86", "")
         )
+        launch_string = UnityEnvironment.environment_launch_check(env_path)
```

Comment:
This isn't quite the same check as in `executable_launcher()`, since we don't do the `strip()` and replace the extensions - should that be moved into `environment_launch_check`?